### ISSUE TITLE
google_biquery_job crashing: cannot create extract job

### ISSUE THREAD

**Issue by rlopezl-c4** [terraform-providers/terraform-provider-google#6364]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version
Terraform v0.12.24

### Affected Resource(s)
* google_bigquery_job

### Terraform Configuration Files

```
resource "google_bigquery_job" "backup_job" {
  for_each = toset(var.table_ids)

  job_id   = var.job_id
  location = var.location

  extract {
    destination_uris = ["${var.backup_gcs_bucket}/${var.dataset_id}/${each.key}"]

    source_table {
      project_id = lookup(var.projects, var.project)
      dataset_id = var.dataset_id
      table_id   = each.key
    }

    destination_format = var.destination_format
    compression        = var.compression
  }

}
```

### Debug Output
https://gist.github.com/rlopezl-c4/ea2d1de20e145cf22e48633581578328

### Panic Output
https://gist.github.com/rlopezl-c4/ea2d1de20e145cf22e48633581578328

### Expected Behavior
Create a functional bigquery job extract data from a table and save it to GCS bucket

### Actual Behavior
Error: Provider produced inconsistent result after apply

When applying changes to google_bigquery_job.backup_job["test_table"],
provider "registry.terraform.io/-/google" produced an unexpected new value for
was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.

### Steps to Reproduce
1. `terraform apply`

**Comment by rlopezl-c4** [comment]
Any news about this? Thank you so much

**Comment by c2thorn** [comment]
Hi @rlopezl-c4, sorry for the delay. I'm not able to reproduce your issue, however from the logs, it seems like you hit some eventual consistency with the job creation. Do you know if the job was actually created after the initial Terraform error?

**Comment by rlopezl-c4** [comment]
Hi,

Nowadays, it's the same error every time I launched the job.

![imagen](https://user-images.githubusercontent.com/63583525/83132183-50172000-a0e1-11ea-8201-6763e6ee9830.png)

The job is created, but the error message in the provider is annoyed:

google_bigquery_job.backup_job["test_table"]: Creating...

Error: Provider produced inconsistent result after apply

When applying changes to google_bigquery_job.backup_job["test_table"],
provider "registry.terraform.io/-/google" produced an unexpected new value for
was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.

[terragrunt] 2020/05/28 12:44:26 Hit multiple errors:
exit status 1

**Comment by c2thorn** [comment]
Hi @rlopezl-c4, thanks for confirming that the job is actually created. I've added asynchronous polling after job creation which usually patches this type of eventual consistency.

The deadline for this fix to show in v3.24 has already passed, so it will likely be a part of v3.25 of the provider. I'd recommend building the provider from the fix commit so you can try it out immediately.
